mass_mailing_custom_unsubscribe/models/mail_mass_mailing.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from itertools import groupby
+
+
+class MailMassMailing(models.Model):
+    _inherit = "mail.mass_mailing"
+
+    def update_opt_out(self, email, list_ids, value):
```

Comment:
@api.multi

Reply:
In fact, it's not needed and we should tend to the contrary: all methods are `@api.multi` by default, and it's going to be removed in v13.